src/modules/sharing/sharing.service.ts

```diff
@@ -1799,4 +1787,67 @@ export class SharingService {
         });
     }
   }
+
+  async changeSharingType(
+    user: User,
+    itemId: Sharing['itemId'],
+    itemType: Sharing['itemType'],
+    type: Sharing['type'],
+  ): Promise<void> {
+    let item: File | Folder;
+
+    if (itemType === 'file') {
+      item = await this.fileUsecases.getByUuid(itemId);
+    } else if (itemType === 'folder') {
+      item = await this.folderUsecases.getByUuid(itemId);
+    }
+    const isUserOwner = item.isOwnedBy(user);
+
+    if (!isUserOwner) {
+      throw new ForbiddenException();
+    }
+
+    const oldSharingType =
+      type === SharingType.Private ? SharingType.Public : SharingType.Private;
+
+    await this.sharingRepository.updateSharing(
```

Comment:
Let me know if we expect to delete the public sharing when switching between modes.
I used update so the previous links can continue being accessible instead of creating a new one.